From e2b34ce7c18f5650fd23cf810251a6630449fa99 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Thu, 7 May 2020 11:05:05 +0200
Subject: [PATCH 1/2] change python/Makefile to point to libfaiss.$(SHLIB_ExT)
 instead of .a

also give definite names to makefile-rules, as otherwise make may
struggle to find make _swigfaiss.so otherwise.
---
 python/Makefile | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/python/Makefile b/python/Makefile
index 2836568..eaa32a4 100644
--- a/python/Makefile
+++ b/python/Makefile
@@ -12,18 +12,18 @@ endif
 all: build
 
 # Also silently generates swigfaiss.py.
-swigfaiss.cpp: swigfaiss.swig ../libfaiss.a
+swigfaiss.cpp: swigfaiss.swig $(PREFIX)/lib/libfaiss$(SHLIB_EXT)
 	$(SWIG) -python -c++ -Doverride= -I../ $(SWIGFLAGS) -o $@ $<
 
 swigfaiss_avx2.cpp: swigfaiss.swig ../libfaiss.a
 	$(SWIG) -python -c++ -Doverride= -module swigfaiss_avx2 -I../ $(SWIGFLAGS) -o $@ $<
 
-%.o: %.cpp
+swigfaiss.o: swigfaiss.cpp
 	$(CXX) $(CPPFLAGS) $(CXXFLAGS) $(CPUFLAGS) $(PYTHONCFLAGS) \
                -I../ -c $< -o $@
 
 # Extension is .so even on OSX.
-_%.so: %.o ../libfaiss.a
+_swigfaiss.so: swigfaiss.o $(PREFIX)/lib/libfaiss$(SHLIB_EXT)
 	$(CXX) $(SHAREDFLAGS) $(LDFLAGS) -o $@ $^ $(LIBS)
 
 build: _swigfaiss.so faiss.py
-- 
2.26.2.windows.1

